sdk/src/packet.rs

```diff
@@ -38,121 +42,143 @@ pub struct Meta {
     pub sender_stake: u64,
 }
 
-#[derive(Clone, Eq)]
-#[repr(C)]
-pub struct Packet {
-    // Bytes past Packet.meta.size are not valid to read from.
-    // Use Packet.data(index) to read from the buffer.
-    buffer: [u8; PACKET_DATA_SIZE],
-    meta: Meta,
-}
+pub trait BasePacket: Default + Clone + Sync + Send + Eq + fmt::Debug {
+    const DATA_SIZE: usize;
+    fn populate_packet<T: Serialize>(&mut self, dest: Option<&SocketAddr>, data: &T) -> Result<()>;
```

Comment:
What is the benefit of this trait as opposed to using _const generics_ for packet data size and then defining some typedefs for `Packet` and `TransactionPacket`?
Something like:
```rust
struct GenericPacket<const N: usize> {
    buffer: [u8; N],
    meta: Meta,
}

type Packet = GenericPacket<PACKET_DATA_SIZE>;
type TransactionPacket = GenericPacket<TRANSACTION_DATA_SIZE>;
```

Reply:
Ok, now it's actually done, along with the removal of `meta()` and `meta_mut()`